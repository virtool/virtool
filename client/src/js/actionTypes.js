/**
 * Created by igboyes on 03/05/17.
 */

const createRequestActionType = (root) => {
    return {
        REQUESTED: `${root}_REQUESTED`,
        SUCCEEDED: `${root}_SUCCEEDED`,
        FAILED: `${root}_FAILED`
    };
};

export const SET_APP_PENDING = "SET_APP_PENDING";
export const UNSET_APP_PENDING = "UNSET_APP_PENDING";

// Jobs
export const WS_UPDATE_JOB = "WS_UPDATE_JOB";
export const WS_REMOVE_JOB = "WS_REMOVE_JOB";
export const FIND_JOBS = createRequestActionType("FIND_JOBS");
export const GET_JOB = createRequestActionType("GET_JOB");
export const CANCEL_JOB = createRequestActionType("CANCEL_JOB");
export const REMOVE_JOB = createRequestActionType("REMOVE_JOB");
export const CLEAR_JOBS = createRequestActionType("CLEAR_JOBS");
export const TEST_JOB = createRequestActionType("TEST_JOB");
export const GET_RESOURCES = createRequestActionType("GET_RESOURCES");
export const GET_CUDA = createRequestActionType("GET_CUDA");

// Samples
export const WS_UPDATE_SAMPLE = "WS_UPDATE_SAMPLE";
export const WS_REMOVE_SAMPLE = "WS_REMOVE_SAMPLE";

export const FIND_SAMPLES = createRequestActionType("FIND_SAMPLES");
export const GET_SAMPLE = createRequestActionType("GET_SAMPLE");
export const UPDATE_SAMPLE = createRequestActionType("UPDATE_SAMPLE");
export const FIND_ANALYSES = createRequestActionType("FIND_ANALYSES");
export const GET_ANALYSIS = createRequestActionType("GET_ANALYSIS");
export const ANALYZE = createRequestActionType("ANALYZE");

// Virus
export const WS_UPDATE_VIRUS = "WS_UPDATE_VIRUS";
export const WS_REMOVE_VIRUS = "WS_REMOVE_VIRUS";

export const FIND_VIRUSES = createRequestActionType("FIND_VIRUSES");
export const GET_VIRUS = createRequestActionType("GET_VIRUS");
export const CREATE_VIRUS = createRequestActionType("CREATE_VIRUS");
export const REMOVE_VIRUS = createRequestActionType("REMOVE_VIRUS");

export const GET_VIRUS_HISTORY = createRequestActionType("GET_VIRUS_HISTORY");

export const SHOW_ADD_ISOLATE = "SHOW_ADD_ISOLATE";
export const SHOW_EDIT_ISOLATE = "SHOW_EDIT_ISOLATE";
export const SHOW_REMOVE_ISOLATE = "SHOW_REMOVE_ISOLATE";
export const HIDE_VIRUS_MODAL = "HIDE_VIRUS_MODAL";

export const ADD_ISOLATE = createRequestActionType("ADD_ISOLATE");
export const EDIT_ISOLATE = createRequestActionType("EDIT_ISOLATE");
export const REMOVE_ISOLATE = createRequestActionType("REMOVE_ISOLATE");

// Indexes
export const WS_UPDATE_INDEX = "WS_UPDATE_INDEX";

export const FIND_INDEXES = createRequestActionType("FIND_INDEXES");
export const GET_INDEX = createRequestActionType("GET_INDEX");
export const CREATE_INDEX = createRequestActionType("CREATE_INDEX");
export const CLEAR_INDEX_ERROR = "CLEAR_INDEX_ERROR";

// Subtraction
export const WS_UPDATE_SUBTRACTION = "WS_UPDATE_SUBTRACTION";
export const WS_REMOVE_SUBTRACTION = "WS_REMOVE_SUBTRACTION";

export const FIND_SUBTRACTIONS = createRequestActionType("FIND_SUBTRACTIONS");
export const GET_SUBTRACTION = createRequestActionType("GET_SUBTRACTION");
export const REMOVE_SUBTRACTION = createRequestActionType("REMOVE_SUBTRACTION");

// Files
export const FIND_FILES = createRequestActionType("FIND_FILES");
export const REMOVE_FILE = createRequestActionType("REMOVE_FILE");
export const UPLOAD_READS = createRequestActionType("UPLOAD_READS");
export const UPLOAD_PROGRESS = "UPLOAD_PROGRESS";
export const HIDE_UPLOAD_OVERLAY = "HIDE_UPLOAD_OVERLAY";



// Account
export const GET_ACCOUNT_REQUESTED = "GET_ACCOUNT_REQUESTED";
export const GET_ACCOUNT_SUCCEEDED = "GET_ACCOUNT_SUCCEEDED";
export const GET_ACCOUNT_FAILED = "GET_ACCOUNT_FAILED";
export const LOGOUT_REQUESTED = "LOGOUT_REQUESTED";
export const LOGOUT_SUCCEEDED = "LOGOUT_SUCCEEDED";

// Administrative setting actionTypes
export const GET_SETTINGS = createRequestActionType("GET_SETTINGS");
export const UPDATE_SETTINGS = createRequestActionType("UPDATE_SETTINGS");

export const SET_SOURCE_TYPE_VALUE = "SET_SOURCE_TYPE_VALUE";
export const SET_CONTROL_READAHEAD_TERM = "SET_CONTROL_READAHEAD_TERM";
export const GET_CONTROL_READAHEAD_REQUESTED = "GET_CONTROL_READAHEAD_REQUESTED";
export const GET_CONTROL_READAHEAD_SUCCEEDED = "GET_CONTROL_READAHEAD_SUCCEEDED";
export const GET_CONTROL_READAHEAD_FAILED = "GET_CONTROL_READAHEAD_FAILED";

// Users
export const LIST_USERS = createRequestActionType("LIST_USERS");
export const SELECT_USER = "SELECT_USER";
export const CHANGE_SET_PASSWORD = "CHANGE_SET_PASSWORD";
export const CHANGE_SET_CONFIRM = "CHANGE_SET_CONFIRM";
export const SET_PASSWORD = createRequestActionType("SET_PASSWORD");
export const CLEAR_SET_PASSWORD = "CLEAR_SET_PASSWORD";
export const SET_FORCE_RESET = createRequestActionType("SET_FORCE_RESET");
export const SET_PRIMARY_GROUP = createRequestActionType("SET_PRIMARY_GROUP");
export const ADD_USER_TO_GROUP = createRequestActionType("ADD_USER_TO_GROUP");
export const REMOVE_USER_FROM_GROUP = createRequestActionType("REMOVE_USER_FROM_GROUP");

// Groups
export const LIST_GROUPS = createRequestActionType("LIST_GROUPS");
