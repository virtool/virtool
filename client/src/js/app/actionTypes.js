/**
 * @module actionTypes
 */

/**
 * Create a special action type used for requests.
 *
 * The request is an object with three properties assigned with action types used for API requests.
 *
 * - REQUESTED
 * - SUCCEEDED
 * - FAILED
 *
 * @param root {string} the root name of the action type
 * @returns {object} a request-style action type
 */
const createRequestActionType = root => ({
    REQUESTED: `${root}_REQUESTED`,
    SUCCEEDED: `${root}_SUCCEEDED`,
    FAILED: `${root}_FAILED`
});

// App
export const PUSH_STATE = "PUSH_STATE";
export const SET_APP_PENDING = "SET_APP_PENDING";
export const UNSET_APP_PENDING = "UNSET_APP_PENDING";

// Dev
export const POST_DEV_COMMAND = createRequestActionType("POST_DEV_COMMAND");

// Account
export const GET_ACCOUNT = createRequestActionType("GET_ACCOUNT");
export const UPDATE_ACCOUNT = createRequestActionType("UPDATE_ACCOUNT");
export const GET_ACCOUNT_SETTINGS = createRequestActionType("GET_ACCOUNT_SETTINGS");
export const UPDATE_ACCOUNT_SETTINGS = createRequestActionType("UPDATE_ACCOUNT_SETTINGS");
export const CHANGE_ACCOUNT_PASSWORD = createRequestActionType("CHANGE_ACCOUNT_PASSWORD");
export const GET_API_KEYS = createRequestActionType("GET_API_KEYS");
export const CREATE_API_KEY = createRequestActionType("CREATE_API_KEY");
export const UPDATE_API_KEY = createRequestActionType("UPDATE_API_KEY");
export const REMOVE_API_KEY = createRequestActionType("REMOVE_API_KEY");
export const CLEAR_API_KEY = "CLEAR_API_KEY";
export const LOGIN = createRequestActionType("LOGIN");
export const LOGOUT = createRequestActionType("LOGOUT");
export const RESET_PASSWORD = createRequestActionType("RESET_PASSWORD");

// Administration Settings
export const GET_SETTINGS = createRequestActionType("GET_SETTINGS");
export const UPDATE_SETTINGS = createRequestActionType("UPDATE_SETTINGS");

// Analysis
export const WS_INSERT_ANALYSIS = "WS_INSERT_ANALYSIS";
export const WS_UPDATE_ANALYSIS = "WS_UPDATE_ANALYSIS";
export const WS_REMOVE_ANALYSIS = "WS_REMOVE_ANALYSIS";
export const FIND_ANALYSES = createRequestActionType("FIND_ANALYSES");
export const ANALYZE = createRequestActionType("ANALYZE");
export const BLAST_NUVS = createRequestActionType("BLAST_NUVS");
export const CLEAR_ANALYSIS = "CLEAR_ANALYSIS";
export const CLEAR_ANALYSES = "CLEAR_ANALYSES";
export const GET_ANALYSIS = createRequestActionType("GET_ANALYSIS");
export const REMOVE_ANALYSIS = createRequestActionType("REMOVE_ANALYSIS");
export const SET_ANALYSIS_ACTIVE_ID = "SET_ANALYSIS_ACTIVE_ID";
export const SET_SEARCH_IDS = "SET_SEARCH_IDS";
export const SET_AODP_FILTER = "ADD_AODP_FILTER";
export const SET_ANALYSIS_SORT_KEY = "SET_ANALYSIS_SORT_KEY";
export const TOGGLE_FILTER_OTUS = "TOGGLE_FILTER_OTUS";
export const TOGGLE_FILTER_ISOLATES = "TOGGLE_FILTER_ISOLATES";
export const TOGGLE_FILTER_ORFS = "TOGGLE_FILTER_ORFS";
export const TOGGLE_FILTER_SEQUENCES = "TOGGLE_FILTER_SEQUENCES";
export const TOGGLE_ANALYSIS_SORT_DESCENDING = "TOGGLE_ANALYSIS_SORT_DESCENDING";
export const TOGGLE_SHOW_PATHOSCOPE_READS = "TOGGLE_SHOW_PATHOSCOPE_READS";

// Caches
export const GET_CACHE = createRequestActionType("GET_CACHE");

// Errors
export const CLEAR_ERROR = "CLEAR_ERROR";

// Files
export const WS_INSERT_FILE = "WS_INSERT_FILE";
export const WS_UPDATE_FILE = "WS_UPDATE_FILE";
export const WS_REMOVE_FILE = "WS_REMOVE_FILE";
export const FIND_FILES = createRequestActionType("FIND_FILES");
export const REMOVE_FILE = createRequestActionType("REMOVE_FILE");
export const UPLOAD = createRequestActionType("UPLOAD");
export const UPLOAD_SAMPLE_FILE = createRequestActionType("UPLOAD_SAMPLE_FILE");
export const UPLOAD_PROGRESS = "UPLOAD_PROGRESS";

// Groups
export const WS_INSERT_GROUP = "WS_INSERT_GROUP";
export const WS_UPDATE_GROUP = "WS_UPDATE_GROUP";
export const WS_REMOVE_GROUP = "WS_REMOVE_GROUP";
export const CHANGE_ACTIVE_GROUP = "CHANGE_ACTIVE_GROUP";
export const LIST_GROUPS = createRequestActionType("LIST_GROUPS");
export const CREATE_GROUP = createRequestActionType("CREATE_GROUP");
export const SET_GROUP_PERMISSION = createRequestActionType("SET_GROUP_PERMISSION");
export const REMOVE_GROUP = createRequestActionType("REMOVE_GROUP");

// HMMs
export const FIND_HMMS = createRequestActionType("FIND_HMMS");
export const GET_HMM = createRequestActionType("GET_HMM");
export const INSTALL_HMMS = createRequestActionType("INSTALL_HMMS");
export const PURGE_HMMS = createRequestActionType("PURGE_HMMS");

// Indexes
export const WS_INSERT_INDEX = "WS_INSERT_INDEX";
export const WS_UPDATE_INDEX = "WS_UPDATE_INDEX";
export const WS_REMOVE_INDEX = "WS_REMOVE_INDEX";
export const WS_INSERT_HISTORY = "WS_INSERT_HISTORY";
export const FIND_INDEXES = createRequestActionType("FIND_INDEXES");
export const GET_INDEX = createRequestActionType("GET_INDEX");
export const GET_UNBUILT = createRequestActionType("GET_UNBUILT");
export const CREATE_INDEX = createRequestActionType("CREATE_INDEX");
export const GET_INDEX_HISTORY = createRequestActionType("GET_INDEX_HISTORY");
export const LIST_READY_INDEXES = createRequestActionType("LIST_READY_INDEXES");

// Jobs
export const WS_INSERT_JOB = "WS_INSERT_JOB";
export const WS_UPDATE_JOB = "WS_UPDATE_JOB";
export const WS_REMOVE_JOB = "WS_REMOVE_JOB";
export const FIND_JOBS = createRequestActionType("FIND_JOBS");
export const GET_JOB = createRequestActionType("GET_JOB");
export const GET_LINKED_JOB = createRequestActionType("GET_LINKED_JOB");
export const CANCEL_JOB = createRequestActionType("CANCEL_JOB");
export const REMOVE_JOB = createRequestActionType("REMOVE_JOB");
export const CLEAR_JOBS = createRequestActionType("CLEAR_JOBS");
export const GET_RESOURCES = createRequestActionType("GET_RESOURCES");

// OTU
export const WS_INSERT_OTU = "WS_INSERT_OTU";
export const WS_UPDATE_OTU = "WS_UPDATE_OTU";
export const WS_REMOVE_OTU = "WS_REMOVE_OTU";
export const FIND_OTUS = createRequestActionType("FIND_OTUS");
export const REFRESH_OTUS = createRequestActionType("REFRESH_OTUS");
export const GET_OTU = createRequestActionType("GET_OTU");
export const CREATE_OTU = createRequestActionType("CREATE_OTU");
export const EDIT_OTU = createRequestActionType("EDIT_OTU");
export const REMOVE_OTU = createRequestActionType("REMOVE_OTU");
export const GET_OTU_HISTORY = createRequestActionType("GET_OTU_HISTORY");
export const ADD_ISOLATE = createRequestActionType("ADD_ISOLATE");
export const EDIT_ISOLATE = createRequestActionType("EDIT_ISOLATE");
export const SET_ISOLATE_AS_DEFAULT = createRequestActionType("SET_ISOLATE_AS_DEFAULT");
export const REMOVE_ISOLATE = createRequestActionType("REMOVE_ISOLATE");
export const ADD_SEQUENCE = createRequestActionType("ADD_SEQUENCE");
export const EDIT_SEQUENCE = createRequestActionType("EDIT_SEQUENCE");
export const REMOVE_SEQUENCE = createRequestActionType("REMOVE_SEQUENCE");
export const REVERT = createRequestActionType("REVERT");
export const UPLOAD_IMPORT = createRequestActionType("UPLOAD_IMPORT");
export const SELECT_ISOLATE = "SELECT_ISOLATE";
export const SHOW_EDIT_OTU = "SHOW_EDIT_OTU";
export const SHOW_REMOVE_OTU = "SHOW_REMOVE_OTU";
export const SHOW_ADD_ISOLATE = "SHOW_ADD_ISOLATE";
export const SHOW_EDIT_ISOLATE = "SHOW_EDIT_ISOLATE";
export const SHOW_REMOVE_ISOLATE = "SHOW_REMOVE_ISOLATE";
export const SHOW_ADD_SEQUENCE = "SHOW_ADD_SEQUENCE";
export const SHOW_EDIT_SEQUENCE = "SHOW_EDIT_SEQUENCE";
export const SHOW_REMOVE_SEQUENCE = "SHOW_REMOVE_SEQUENCE";
export const HIDE_OTU_MODAL = "HIDE_OTU_MODAL";

// Tasks
export const WS_INSERT_TASK = "WS_INSERT_TASK";
export const WS_UPDATE_TASK = "WS_UPDATE_TASK";
export const LIST_TASKS = createRequestActionType("LIST_TASKS");
export const GET_TASK = createRequestActionType("GET_TASK");

// Refs
export const WS_INSERT_REFERENCE = "WS_INSERT_REFERENCE";
export const WS_UPDATE_REFERENCE = "WS_UPDATE_REFERENCE";
export const WS_REMOVE_REFERENCE = "WS_REMOVE_REFERENCE";
export const FIND_REFERENCES = createRequestActionType("FIND_REFERENCES");
export const GET_REFERENCE = createRequestActionType("GET_REFERENCE");
export const CREATE_REFERENCE = createRequestActionType("CREATE_REFERENCE");
export const EDIT_REFERENCE = createRequestActionType("EDIT_REFERENCE");
export const REMOVE_REFERENCE = createRequestActionType("REMOVE_REFERENCE");
export const IMPORT_REFERENCE = createRequestActionType("IMPORT_REFERENCE");
export const CLONE_REFERENCE = createRequestActionType("CLONE_REFERENCE");
export const REMOTE_REFERENCE = createRequestActionType("REMOTE_REFERENCE");
export const ADD_REFERENCE_USER = createRequestActionType("ADD_REFERENCE_USER");
export const EDIT_REFERENCE_USER = createRequestActionType("EDIT_REFERENCE_USER");
export const REMOVE_REFERENCE_USER = createRequestActionType("REMOVE_REFERENCE_USER");
export const ADD_REFERENCE_GROUP = createRequestActionType("ADD_REFERENCE_GROUP");
export const EDIT_REFERENCE_GROUP = createRequestActionType("EDIT_REFERENCE_GROUP");
export const REMOVE_REFERENCE_GROUP = createRequestActionType("REMOVE_REFERENCE_GROUP");
export const CHECK_REMOTE_UPDATES = createRequestActionType("CHECK_REMOTE_UPDATES");
export const UPDATE_REMOTE_REFERENCE = createRequestActionType("UPDATE_REMOTE_REFERENCE");

// Samples
export const WS_INSERT_SAMPLE = "WS_INSERT_SAMPLE";
export const WS_UPDATE_SAMPLE = "WS_UPDATE_SAMPLE";
export const WS_REMOVE_SAMPLE = "WS_REMOVE_SAMPLE";
export const FIND_SAMPLES = createRequestActionType("FIND_SAMPLES");
export const FIND_READ_FILES = createRequestActionType("FIND_READ_FILES");
export const GET_SAMPLE = createRequestActionType("GET_SAMPLE");
export const CREATE_SAMPLE = createRequestActionType("CREATE_SAMPLE");
export const UPDATE_SAMPLE = createRequestActionType("UPDATE_SAMPLE");
export const UPDATE_SAMPLE_RIGHTS = createRequestActionType("UPDATE_SAMPLE_RIGHTS");
export const REMOVE_SAMPLE = createRequestActionType("REMOVE_SAMPLE");
export const SELECT_SAMPLE = "SELECT_SAMPLE";
export const DESELECT_SAMPLES = "DESELECT_SAMPLES";
export const CLEAR_SAMPLE_SELECTION = "CLEAR_SAMPLE_SELECTION";
export const SHOW_REMOVE_SAMPLE = "SHOW_REMOVE_SAMPLE";
export const HIDE_SAMPLE_MODAL = "HIDE_SAMPLE_MODAL";
export const REPLACE_LEGACY_FILES = createRequestActionType("REPLACE_LEGACY_FILES");

// Subtraction
export const WS_INSERT_SUBTRACTION = "WS_INSERT_SUBTRACTION";
export const WS_UPDATE_SUBTRACTION = "WS_UPDATE_SUBTRACTION";
export const WS_REMOVE_SUBTRACTION = "WS_REMOVE_SUBTRACTION";
export const FIND_SUBTRACTIONS = createRequestActionType("FIND_SUBTRACTIONS");
export const SHORTLIST_SUBTRACTIONS = createRequestActionType("LIST_SUBTRACTION_IDS");
export const GET_SUBTRACTION = createRequestActionType("GET_SUBTRACTION");
export const CREATE_SUBTRACTION = createRequestActionType("CREATE_SUBTRACTION");
export const EDIT_SUBTRACTION = createRequestActionType("UPDATE_SUBTRACTION");
export const REMOVE_SUBTRACTION = createRequestActionType("REMOVE_SUBTRACTION");

// Users
export const WS_INSERT_USER = "WS_INSERT_USER";
export const WS_UPDATE_USER = "WS_UPDATE_USER";
export const WS_REMOVE_USER = "WS_REMOVE_USER";
export const FIND_USERS = createRequestActionType("FIND_USERS");
export const GET_USER = createRequestActionType("GET_USER");
export const CREATE_USER = createRequestActionType("CREATE_USER");
export const CREATE_FIRST_USER = createRequestActionType("CREATE_FIRST_USER");
export const EDIT_USER = createRequestActionType("EDIT_USER");
export const REMOVE_USER = createRequestActionType("REMOVE_USER");

// Updates
export const WS_UPDATE_STATUS = "WS_UPDATE_STATUS";
export const GET_SOFTWARE_UPDATES = createRequestActionType("GET_SOFTWARE_UPDATES");
export const INSTALL_SOFTWARE_UPDATES = createRequestActionType("INSTALL_SOFTWARE_UPDATES");

//Labels
export const UPDATE_LABEL = createRequestActionType("UPDATE_LABEL");
export const LIST_LABELS = createRequestActionType("LIST_LABELS");
export const CREATE_LABEL = createRequestActionType("CREATE_LABEL");
export const REMOVE_LABEL = createRequestActionType("REMOVE_LABEL");
