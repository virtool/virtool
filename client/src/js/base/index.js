import { AlgorithmSelect } from "./AlgorithmSelect";
import { AutoProgressBar, ProgressBar } from "./ProgressBar";
import { bsStyles } from "./utils";
import { Checkbox } from "./Checkbox";
import { Flex, FlexItem } from "./Flex";
import { Help } from "./Help";
import { Icon } from "./Icon";
import { Identicon } from "./Identicon";
import { Input } from "./Input";
import { InputError } from "./InputError";
import { InputSave } from "./InputSave";
import { ListGroupItem } from "./ListGroupItem";
import { NotFound } from "./NotFound";
import { Panel } from "./Panel";
import { Radio } from "./Radio";
import { RelativeTime } from "./RelativeTime";
import { RemoveBanner } from "./RemoveBanner";
import { RemoveModal } from "./RemoveModal";
import { SaveButton } from "./SaveButton";
import { ScrollList } from "./ScrollList";
import { Tooltip } from "./Tooltip";
import { UploadBar } from "./UploadBar";
import { ViewHeader } from "./ViewHeader";

export * from "./Alert";
export * from "./Attribution";
export * from "./Badge";
export * from "./Box";
export * from "./Breadcrumb";
export * from "./Breakpoints";
export * from "./Button";
export * from "./ButtonGroup";
export * from "./ButtonToolbar";
export * from "./Code";
export * from "./Container";
export * from "./ExternalLink";
export * from "./Label";
export * from "./Loader";
export * from "./LoadingPlaceholder";
export * from "./Logo";
export * from "./Modal";
export * from "./NoneFound";
export * from "./Table";
export * from "./SectionHeader";
export * from "./SlashList";
export * from "./TabLink";
export * from "./Tabs";
export * from "./Toolbar";
export * from "./ViewHeader";
export * from "./SelectBox";

export {
    AlgorithmSelect,
    AutoProgressBar,
    bsStyles,
    Checkbox,
    Flex,
    FlexItem,
    Help,
    Icon,
    Identicon,
    Input,
    InputError,
    InputSave,
    ListGroupItem,
    NotFound,
    Panel,
    ProgressBar,
    Radio,
    RelativeTime,
    RemoveBanner,
    RemoveModal,
    SaveButton,
    ScrollList,
    Tooltip,
    UploadBar,
    ViewHeader
};
