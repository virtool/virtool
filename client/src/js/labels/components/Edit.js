import { find, get } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { Modal, ModalBody, ModalHeader } from "../../base";
import { routerLocationHasState } from "../../utils/utils";
import { pushState } from "../../app/actions";
import { updateLabel } from "../actions";
import { LabelForm } from "./Form";

const getInitialState = ({ labelName, color, description }) => ({
    labelName: labelName || "",
    color: color || "",
    description: description || "",
    errorName: "",
    errorColor: ""
});

export class Edit extends React.Component {
    constructor(props) {
        super(props);
        this.state = getInitialState(this.props);
    }

    handleChange = e => {
        const { name, value } = e.target;
        this.setState({
            [name]: value,
            error: ""
        });
    };

    handleColorSelection = color => {
        this.setState({ color });
    };

    handleModalEnter = () => {
        this.setState(getInitialState(this.props));
    };

    handleSubmit = e => {
        e.preventDefault();
        if (this.state.labelName === "") {
            this.setState({ errorName: "Please enter a label name" });
        } else if (this.state.color === "") {
            this.setState({ errorColor: "Please select a color" });
        } else {
            this.props.onSubmit(this.props.id, this.state.labelName, this.state.description, this.state.color);
            this.props.onHide();
        }
    };

    render() {
        const { labelName, description, color, errorName, errorColor } = this.state;
        return (
            <Modal label="Edit Label" show={this.props.show} onEnter={this.handleModalEnter} onHide={this.props.onHide}>
                <ModalHeader>Edit a label</ModalHeader>
                <ModalBody>
                    <LabelForm
                        color={color}
                        description={description}
                        name={labelName}
                        errorColor={errorColor}
                        errorName={errorName}
                        onChange={this.handleChange}
                        onColorChange={this.handleColorSelection}
                        onSubmit={this.handleSubmit}
                    />
                </ModalBody>
            </Modal>
        );
    }
}

export const mapStateToProps = (state, ownProps) => {
    const { id } = ownProps;
    const document = find(state.labels.list, { id });
    if (document) {
        const { color, description, name } = document;
        return {
            id,
            color,
            description,
            labelName: name,
            show: routerLocationHasState(state, "editLabel"),
            error: get(state, "errors.EDIT_SAMPLE_ERROR.message", "")
        };
    }

    return {
        show: routerLocationHasState(state, "editLabel")
    };
};

export const mapDispatchToProps = dispatch => ({
    onSubmit: (id, name, description, color) => {
        dispatch(updateLabel(id, name, description, color));
    },
    onHide: () => {
        dispatch(pushState({ editLabel: false }));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Edit);
