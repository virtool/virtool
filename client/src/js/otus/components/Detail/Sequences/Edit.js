import { find } from "lodash-es";
import React, { useCallback } from "react";
import { connect } from "react-redux";
import { Modal, ModalHeader } from "../../../../base";
import { clearError } from "../../../../errors/actions";
import { editSequence, hideOTUModal } from "../../../actions";
import { getSequences, getTargetName } from "../../../selectors";
import { SequenceForm } from "./Form";

const EditSequence = ({
    accession,
    dataType,
    definition,
    host,
    id,
    sequence,
    targetName,
    otuId,
    isolateId,
    onHide
}) => {
    const handleSubmit = useCallback(
        ({ accession, definition, host, sequence, segment, targetName }) =>
            this.props.onSave(
                this.props.otuId,
                this.props.isolateId,
                this.props.id,
                accession,
                definition,
                host,
                sequence,
                segment,
                targetName
            ),
        [otuId, isolateId, id]
    );

    return (
        <Modal label="Edit Sequence" show={!!id} onHide={onHide}>
            <ModalHeader>Edit Sequence</ModalHeader>
            <SequenceForm
                key={`edit_${id}`}
                accession={accession}
                definition={definition}
                host={host}
                sequence={sequence}
                targetName={targetName}
                dataType={dataType}
                onSubmit={handleSubmit}
            />
        </Modal>
    );
};

const mapStateToProps = state => {
    const id = state.otus.editSequence;

    if (id) {
        const sequences = getSequences(state);
        const { accession, definition, host, segment, sequence } = find(sequences, { id });

        return {
            id,
            accession,
            definition,
            host,
            sequence,
            segment,
            targetName: getTargetName(state),
            dataType: state.references.detail.data_type
        };
    }

    return {};
};

const mapDispatchToProps = dispatch => ({
    onHide: () => {
        dispatch(hideOTUModal());
    },

    onSave: (otuId, isolateId, sequenceId, accession, definition, host, sequence, segment) => {
        dispatch(editSequence(otuId, isolateId, sequenceId, accession, definition, host, sequence, segment));
    },

    onClearError: error => {
        dispatch(clearError(error));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(EditSequence);
