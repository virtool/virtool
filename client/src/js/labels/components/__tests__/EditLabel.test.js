import { Edit, mapDispatchToProps } from "../Edit";
import { Modal } from "../../../base";
import { PUSH_STATE } from "../../../app/actionTypes";

describe("<Edit>", () => {
    let props, state;
    beforeEach(() => {
        props = {
            id: "1",
            labelName: "FooBar",
            description: "BarFoo",
            color: "#3C8786",
            show: false,
            onHide: jest.fn(),
            onSubmit: jest.fn()
        };
        state = {
            labelName: "FooBar",
            description: "BarFoo",
            color: "#3C8786",
            error: "",
            errorName: "",
            errorColor: ""
        };
    });

    it("should render when [show=false]", () => {
        const wrapper = shallow(<Edit {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [show=true]", () => {
        props.show = true;
        const wrapper = shallow(<Edit {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should update color on handleColorSelection", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({ ...state });
        wrapper.instance().handleColorSelection("#000000");
        expect(wrapper.state()).toEqual({ ...state, color: "#000000" });
    });

    it("should update name on handleChange", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({ ...state });
        const e = {
            target: {
                name: "labelName",
                value: "TestName",
                error: ""
            }
        };
        wrapper.instance().handleChange(e);
        expect(wrapper.state()).toEqual({ ...state, labelName: "TestName" });
    });

    it("should return onSubmit() in props", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({ ...state });
        const e = {
            preventDefault: jest.fn()
        };
        wrapper.instance().handleSubmit(e);
        expect(props.onSubmit).toHaveBeenCalledWith("1", "FooBar", "BarFoo", "#3C8786");
    });

    it("should submit onSubmit() with no description", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({ ...state, description: "" });
        const e = {
            preventDefault: jest.fn()
        };
        wrapper.instance().handleSubmit(e);
        expect(props.onSubmit).toHaveBeenCalledWith("1", "FooBar", "", "#3C8786");
    });

    it("should display error when forms submit with no name", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({ ...state, labelName: "" });
        const e = {
            preventDefault: jest.fn()
        };
        wrapper.instance().handleSubmit(e);
        expect(wrapper.state()).toEqual({ ...state, labelName: "", errorName: "Please enter a label name" });
    });

    it("should display error when forms submit with no color", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({ ...state, color: "" });
        const e = {
            preventDefault: jest.fn()
        };
        wrapper.instance().handleSubmit(e);
        expect(wrapper.state()).toEqual({ ...state, color: "", errorColor: "Please select a color" });
    });

    it("should return onHide() in props", () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.find(Modal).props().onHide();
        expect(props.onHide).toHaveBeenCalled();
    });
});

describe("mapDispatchToProps()", () => {
    let dispatch;
    let props;

    beforeEach(() => {
        dispatch = jest.fn();
        props = mapDispatchToProps(dispatch);
    });

    it("should return onSubmit in props", () => {
        const labelId = "1";
        const name = "FooBar";
        const description = "BarFoo";
        const color = "#000000";
        props.onSubmit(labelId, name, description, color);
        expect(dispatch).toHaveBeenCalledWith({
            type: "UPDATE_LABEL_REQUESTED",
            labelId,
            name,
            description,
            color
        });
    });

    it("should return onHide in props", () => {
        props.onHide();
        expect(dispatch).toHaveBeenCalledWith({
            type: PUSH_STATE,
            state: {
                editLabel: false
            }
        });
    });
});
