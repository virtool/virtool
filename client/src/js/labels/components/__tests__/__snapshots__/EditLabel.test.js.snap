// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit> should render when [show=false] 1`] = `
<Modal
  label="Edit Label"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={false}
>
  <ModalHeader>
    Edit a label
  </ModalHeader>
  <Modal__ModalBody>
    <LabelForm
      color="#3C8786"
      description="BarFoo"
      errorColor=""
      errorName=""
      name="FooBar"
      onChange={[Function]}
      onColorChange={[Function]}
      onSubmit={[Function]}
    />
  </Modal__ModalBody>
</Modal>
`;

exports[`<Edit> should render when [show=true] 1`] = `
<Modal
  label="Edit Label"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Edit a label
  </ModalHeader>
  <Modal__ModalBody>
    <LabelForm
      color="#3C8786"
      description="BarFoo"
      errorColor=""
      errorName=""
      name="FooBar"
      onChange={[Function]}
      onColorChange={[Function]}
      onSubmit={[Function]}
    />
  </Modal__ModalBody>
</Modal>
`;
